### PR TITLE
Add Move language

### PR DESCRIPTION
**Pull request by uvd**
## Description

Adds [Move](https://github.com/move-language/move) programming language with the .move extension

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Amove+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - Main.move    wrote it myself  on this pr 
      - [[Vector.move)]](https://github.com/starcoinorg/starcoin-framework/blob/main/sources/Vector.move)
    - Sample license(s): mit
  - [x] I have included a syntax highlighting grammar: [[vscode-move-syntax]](https://github.com/damirka/vscode-move-syntax)
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by uvd** [comment]
hi, this ci not successful , but  It's   seem  not my code ,I need help ,or tell me how to do

**Comment by uvd** [comment]
hi，i'm change my code ，help re-run  ci

File: samples/Move/Main.move [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    use StarcoinFramework::Account;
+    use StarcoinFramework::Signer;
+    use StarcoinFramework::Token;
+    use StarcoinFramework::Event;
+    use SFC::PseudoRandom;
+
+    struct Bank<phantom T: store> has store, key {
+        bank: Token::Token<T>
+    }
+
```
Comment (jolestar) [inline samples/Move/Main.move:12]: I think we should use a more simple example, mainly used to show the syntax characteristics of Move.

Comment (uvd) [reply samples/Move/Main.move:12]: 👌🏻

**Comment by lildude** [comment]
Please update the OP to point to where you got the new sample from and its licence.

**Comment by lildude** [comment]
The test failures tell you what you need to do to fix things. I recommend running the tests locally (`bundle exec rake test`) until they pass and then push your changes.

**Comment by uvd** [comment]
> The test failures tell you what you need to do to fix things. I recommend running the tests locally (`bundle exec rake test`) until they pass and then push your changes.

hi   i change some code order  ,  I'm try local    all test pass , can try  run CI again  ,3q

`Finished in 20.045657s, 79.7180 runs/s, 1623.5936 assertions/s.

1598 runs, 32546 assertions, 0 failures, 0 errors, 0 skips`

**Comment by uvd** [comment]
o my god ~  It's too strict , I formatted it a little bit for licence again ，I'm sorry ，Commit the code for the first time for this ，I‘m changed it  ,try it again 😭😭

**Comment by uvd** [comment]
>     3q (thanks) you very much

**Comment by uvd** [comment]
I would like to ask when it will be released to Github

**Comment by lildude** [comment]
Probably some time during the week of 11 July

**Comment by damirka** [comment]
Hey @lildude, thanks for merging it in! How does the syntax get updated? Is it auto-fetched from the repo on tags/pushes? More specifically - what do I need to do to fix current errors in syntax?

**Comment by lildude** [comment]
> How does the syntax get updated?

I manually update all the submodules whenever I make an update ([step 3](https://github.com/github/linguist/blob/master/docs/releasing.md)) so go ahead and make your fixes and they'll be pulled in the next time I make a release. I made a new one yesterday so another won't be for a few months unless something critical comes up that requires a release sooner.

**Comment by damirka** [comment]
Thank you for quick response @lildude. Yes, if that it possible, I would ask you to do it sooner. Only half our ecosystem migrated to the newer syntax, and their code got wrong highlighting. And they were the ones who started the process, so it feels completely unfair. 

I will make sure that the we don't allow mistakes like this one in the future. Honestly didn't even know how it works until @jolestar and @uvd made it happen.

**Comment by uvd** [comment]
I have a question, why can't  find project be searched by language tag   ，Are there any rules that can be indexed      https://github.com/search?q=language%3AMove&type=Repositories&ref=advsearch&l=Move&l=

**Comment by lildude** [comment]
> I have a question, why can't find project be searched by language tag ，Are there any rules that can be indexed https://github.com/search?q=language%3AMove&type=Repositories&ref=advsearch&l=Move&l=

See my previous response at https://github.com/github/linguist/issues/5952#issuecomment-1168505070

**Comment by uvd** [comment]
3q

**Pull request merged**